### DISCUSSION TITLE
Need help getting started - first time doing this

### DISCUSSION THREAD

**Discussion by talormanda** [tronikos/esphome-magnetometer-water-gas-meter#37] [Q&A] [answered]
Hello! First time doing anything really "custom" like this for Home Assistant. I wanted to make sure I was getting the correct parts before I begin my journey. I was hoping to gain some knowledge from those who have done this before so I know I am headed in the right direction.

**My meter:**
![20241001_155718](https://github.com/user-attachments/assets/3e8baf1b-dc35-447e-8df0-e27a91f76057)
![20241003_222943](https://github.com/user-attachments/assets/7a0e131b-64df-4e98-b647-3aedafd88c42)

**From what I understand, I need to get:**
1. ESP8266 or ESP32 --> Which one is better to get? Any suggestions like a direct link would be lovely!
2. QMC5883L or HMC5883L --> Which one is better to get? Any suggestions like a direct link would be lovely!

Is there anything I can build where I don't have to solder? Not the end of the world, but just wondering. My meter is indoors, so I don't have to worry about water damage.

My water meter does react to magnets because I can place a small magnet on it, and when I use water, it physically moves around on the top. If there's anything else I need to know, please inform me if you could.

**Comment by tronikos** [comment]
I believe your posted this in the wrong repository. This isn't using a camera. It's using a magnetometer.

According to the flume meter website your meter is compatible so you should be able to use this project.

**Reply by talormanda** [reply]
> I believe your posted this in the wrong repository. This isn't using a camera. It's using a magnetometer.
> 
> According to the flume meter website your meter is compatible so you should be able to use this project.

wooops. let me change my links. i used the wrong ones.

**Reply by tronikos** [reply]
And if you want to use this project, the parts you listed are wrong. See the readme for what you need. If you don't want to solder there are such parts. I don't have them in the readme but I think someone posted them here.

**Reply by talormanda** [reply]
Okay I guess I never did compile a list for this project, I thought I did, but I can't find it now.

**Reply by talormanda** [reply]
Apologies!

Okay, I updated the post. Here is what I was wondering:

**From what I understand, I need to get:**
1. ESP8266 or ESP32 --> Which one is better to get? Any suggestions like a direct link would be lovely!
2. QMC5883L or HMC5883L --> Which one is better to get? Any suggestions like a direct link would be lovely!

Is there anything I can build where I don't have to solder? Not the end of the world, but just wondering. My meter is indoors, so I don't have to worry about water damage.

**Reply by tronikos** [reply]
I recommend any ESP32 and any QMC5883L. Just search for them at Amazon, eBay, AliExpress etc. If you don't want to solder, I know there are parts but I don't remember their names. I think someone posted them here.

**Reply by talormanda** [reply]
> I recommend any ESP32 and any QMC5883L. Just search for them at Amazon, eBay, AliExpress etc. If you don't want to solder, I know there are parts but I don't remember their names. I think someone posted them here.

I wasn't sure if anyone had any specific issues with one part or another, and to avoid one :-) but if no issues with anything, that's fine.

**Reply by tronikos** [reply]
Yeah anything should work. I got the QMC5883L for like $0.30 and ESP8266 for $0.90 on eBay.

**Reply by talormanda** [reply]
Do those ESP boards just plug in via micro-USB? And then you just connect the magnet board to it, and then the ESP board powers that, and then you just physically stick it to your water meter and --> all good?

**Comment by tronikos** [comment]
Yes micro USB for power, solder wires between ESP and magnetometer, stick the magnetometer on the side of the meter.

**Reply by talormanda** [reply]
Really, the side will work? It doesn't have to be on top?

**Comment by talormanda** [comment]
@tronikos - would these two things work?

1. https://shop.m5stack.com/products/atom-lite-esp32-development-kit
2. https://www.ebay.com/itm/176475984522

if that ESP32 works, wondering how it connects because it has a nice connector on it, which is why I would want to use that. thoughts?

**Reply by talormanda** [reply]
Tagging you again @tronikos in case you missed this. I found this picture, so, this ESP32 device I already have should work right?

It looks like it has the same SCL and SDA spots. Thoughts?

![image](https://github.com/user-attachments/assets/1fe4f5fb-3487-4ea4-aaae-b87b9e0c1fa0)

**Reply by tronikos** [reply]
Yes it should work

**Reply by talormanda** [reply]
Is there a certain change to the code that is needed for this? Like, do I just have to change the pinout lines to direct to G21 and G25?

Also, if I use a pin header, it looks like I can directly plug this ESP32 into the QMC5883L after I solder it to the QMC5883L, bypassing the need for CAT6 cables. It should still work all the same right?

Appreciate your help so far.

**Reply by tronikos** [reply, marked as answer]
Per README, update the substitutions so that the pins match the actual ones you used. 

Yes it should work. It might only be hard to mount the magnetometer together with the ESP device. If the wires are short, you can use any wire you want, no need for CAT6.

**Reply by talormanda** [reply]
Thanks! I'll mark this as complete since I believe that's everything for now.
